crypto/src/hash.rs

```diff
@@ -98,6 +97,16 @@ macro_rules! define_hash {
             }
         }
 
+        impl ::std::fmt::Debug for $name {
+            fn fmt(&self, f: &mut ::std::fmt::Formatter<'_>) -> ::std::fmt::Result {
+                // TODO - TE-373: with b58 this could be done without the need
+                // to perform a heap allocation.
+                f.debug_tuple(stringify!($name))
+                    .field(&self.to_base58_check())
```

Comment:
@tizoc @akoptelov 
and I was thinking about caching `to_base58_check`, I dont know if it is worth and how expensive is base58 calculation